pytorch_lightning/core/memory.py

```diff
@@ -252,7 +269,23 @@ def __str__(self):
             arrays.append(["In sizes", self.in_sizes])
             arrays.append(["Out sizes", self.out_sizes])
 
-        return _format_summary_table(*arrays)
+        if self._mode == ModelSummary.MODE_FULL:
+            # Ignore ModuleList and Sequential since they show the sum of parameters of its contents
+            modules_to_ignore = ["ModuleList", "Sequential"]
+            param_nums = [layer.num_parameters for i, layer in enumerate(self._layer_summary.values())
+                          if self.layer_types[i] not in modules_to_ignore]
+            trainable_param_nums = [layer.num_trainable_parameters
+                                    for i, layer in enumerate(self._layer_summary.values())
+                                    if self.layer_types[i] not in modules_to_ignore]
+
+            total_params = sum(param_nums)
+            total_trainable_params = sum(trainable_param_nums)
```

Comment:
Let's not do this per layer. 
Can we just run your function `num_trainable_parameters` on the root module and show that count at the end?

Reply:
One question @awaelchli, what would be better to do regarding the test that is currently failing? I can think of 2 options:
1 - keep the default value of `named_modules` in `summarize` to `None`, then inside check `if named_modules is None` then use `self.named_modules`.
2 - change the test code to give `self.named_modules` as a parameter to `summarize`.

I prefer the 1st option, what do you think?

Reply:
no let's not change the signature of summarize, if possible please.
Let's just do the summary with the root module, regardless of the mode. this doesn't have to be done by layer.

Reply:
(regardless of the display mode, the total number of parameters and total number of non-trainable parameters is always the same. therefore the implementation should not depend on the mode.

Reply:
So I must create a function that is a copy of the `summarize` function, but always does only in the root module right? Because the `summarize` function uses internally the `named_modules`, which has different content depending on the selected mode

Reply:
Isn't it just this?

```python
trainable_parameters = sum(np.prod(p.shape) for p in self._model.parameters() if p.requires_grad)
total_parameters = sum(np.prod(p.shape) for p in self._model.parameters())
non_trainable_parameters = total_parameters - trainable_parameters
```

what am I missing?

Reply:
Oh, now I get it. I was still thinking of using `LayerSummary` inside. Updating the code now

Reply:
@george-gca I was away for a while, sorry. Thanks for addressing my comments. 
Great PR!